src/main/webapp/2Dscript.js

```diff
@@ -80,7 +175,418 @@ function addCenterControl(controlDiv) {
+      graphData(dataEntries, id);
+    }
+  } else {
+    // In this case the box has become unchecked. We want to remove this graph.
+    // We can only remove this box if it has been generated before and then can
+    // be removed appropriately.
+    const toRemove = runs[id].map;
+    if (toRemove) {
+      toRemove.setMap(null);
+    }
+  }
+}
+
+/**
+ * Generates the header row of a table from an array of column names.
+ * @param {Array<string>} headerRowText
+ * @return {HTMLElement}
+ */
+function generateHeaderRow(headerRowText) {
+  const headerRow = document.createElement('tr');
+  headerRowText.foreach((headerName) => {
+    const currentColumn = document.createElement('th');
+    currentColumn.innerText = headerName;
+    headerRow.appendChild(currentColumn);
+  });
+  return headerRow;
+}
+/**
+ * Generates a table entry for displaying a runId.
+ * @param {string} runId
+ * @return {HTMLElement}
+ */
+function generateKeyEntry(runId) {
+  const keyEntry = document.createElement('td');
+  keyEntry.innerText = runId;
+  return keyEntry;
+}
+
+/**
+ * Generate a checkbox with appropriate event listener for insertion into a
+ * table.
+ * @param {string} runId
+ * @return {HTMLElement}
+ */
+function generateCheckBoxEntry(runId) {
+  const checkBoxEntry = document.createElement('td');
+  const input = document.createElement('input');
+  input.type = 'checkbox';
+  input.id = runId;
+  input.addEventListener('click', showPoseData);
+  checkBoxEntry.appendChild(input);
+  return checkBoxEntry;
+}
+
+/**
+ * Fetch the data and graph it. Necessary to include the graphing in this method
+ * so the asynchronous nature of the fetch operation isn't ignored.
+ * @param {string} runId
+ */
+function fetchAndGraphData(runId) {
+  fetch('/getrun?id=' + runId + '&dataType=pose')
+      .then((response) => response.json())
+      .then((data) => dataEntries = data).then(() => {
+        runs[runId].data = dataEntries;
+        graphData(dataEntries, runId);
+      });
+}
+
+/**
+ * Create a line using the given data entries, plot it on the map and save it to
+ * the cache.
+ * @param {Array<PoseData>} dataEntries
+ * @param {string} runId
+ */
+function graphData(dataEntries, runId) {
+  const toGraph = plotLine(dataEntries);
+  runs[runId].map = toGraph;
+  toGraph.setMap(map);
+}
+
+/**
+ * Insert all the elements in the array into the current table's row.
+ * @param {HTMLElement} currentRow
+ * @param {Array<string>} columnElements
+ */
+function updateRow(currentRow, columnElements) {
+  columnElements.foreach((currentElement) => currentRow.appendChild(currentElement));  // eslint-disable-line
+}
+
+
+/**
+ * Generate a polyline from the given data points and return it.
+ * @param {Array<Point>} dataEntries
+ * @return {google.maps.Polyline}
+ */
+function plotLine(dataEntries) {
+  const currentLine = [];
+  for (point of dataEntries) {
+    currentLine.push({lat: point.lat, lng: point.lng});
+  }
```

Comment:
I think you can do
`const currentLine = [...dataEntries];`
or
`const currentLine = dataEntries.slice();`
instead.

Reply:
This was a mistake in the type annotation, where dataEntries is of type PoseData not Point and this loop removes those extra parameters.

Reply:
I fixed this error in the type annotation